happytransformer/mwp/trainer.py

```diff
@@ -1,14 +1,67 @@
-from happytransformer.happy_trainer import HappyTrainer
-from happytransformer.mwp.default_args import ARGS_MWP_TRAIN
+"""
+Fine-tuning for masked word prediction models.
 
+Based on the tutorial found here:
+https://github.com/huggingface/transformers/blob/master/examples/pytorch/language-modeling/run_mlm.py
+"""
+
+from transformers import DataCollatorForLanguageModeling
+from happytransformer.happy_trainer import HappyTrainer, EvalResult
+from datasets import load_dataset
+from happytransformer.fine_tuning_util import preprocess_concatenate
 
 class WPTrainer(HappyTrainer):
+    """
+    Trainer class for HappyWordPrediction
+    """
 
-    def train(self, input_filepath, args=ARGS_MWP_TRAIN):
-        raise NotImplementedError()
+    def train(self, input_filepath, args):
+        datasets = load_dataset("text", data_files={"train": input_filepath})
+
+        if args["line-by-line"]:
+            tokenized_datasets = self._preprocess_line_by_line(self.tokenizer, datasets, args)
+        else:
+            tokenized_datasets = preprocess_concatenate(self.tokenizer, datasets, args, True)
+
+        data_collator = DataCollatorForLanguageModeling(tokenizer=self.tokenizer,
+                                                        mlm_probability=args['mlm_probability'])
+
+        self._run_train(tokenized_datasets['train'], args, data_collator)
+
+
+    def eval(self, input_filepath, args):
```

Comment:
Shouldn't we be using `kwargs` here?

Reply:
I don't think so. In this case, the args value must contain predefined keys. 

I think **kwargs is more for when there is an arbitrary number of keys. 

Do you agree?

Reply:
Yes, but a dictionary is undesirable for the same reason that `**kwargs` is undesirable.

Consider the code 

```
wp_trainer.train('filepath.csv',{'line-by-lin':True})
```

It silently fails, which does not happen if the method looked like

```
    def train(self, input_filepath, line_by_line=False)
```

Reply:
The code would look really messy if we added a parameter within wp_trainer.eval()  and wp_trainer.train()  for each value. 

I suppose we can let the user input a dictionary (so no breaking changes). Then, we can copy all of the values over to a dataclass. If they do not include a key within the inputted dictionary,  then the default value for the key will be used. Finally, we pass the dataclass to the wp_trainer.eval() method. We  can also add a type checks to ensure that the inputted dataclass is of the proper type, so then we don't have to worry about it not including required keys. 

This change will apply to all HappyTransformer.eval()/.train()/test() methods. 

This functionality would be convenient for users because they may just want to edit a couple values. So then, they would no longer be required to copy all of ARGS_WP_TRAIN, ARGS_WP_EVAl etc dictionaries just to make one small change. They could input something like {"mlm_probability": 0.1} for the HappyTransformer.eval()/.train()/test() args parameter  

Does this sound good to you?

Reply:
The dictionary solution sounds good internally but I still think `**kwargs` is better.

```
wp_trainer.train('filepath.csv',line_by_line=True)
```

is going to be easier for users to understand (and more pythonic) than

```
wp_trainer.train('filepath.csv',{'line-by-line':True})
```

Reply:
See be2a1df

To avoid a breaking change, I kept the input as a dictionary called "args." However, now this dictionary may only contain a subset of the keys and the unused keys will be assigned their default values. This functionality is the same as using "**kwargs" except the "**" is not used. 

I then went onto convert the settings into a dataclass before passing it to wp_trainer.eval().